driver-core/src/test/functional/com/mongodb/internal/connection/SocketStreamHelperSpecification.groovy

```diff
@@ -55,10 +55,18 @@ class SocketStreamHelperSpecification extends Specification {
 
         // If the Java 11+ extended socket options for keep alive probes are available, check those values.
         if (Arrays.stream(ExtendedSocketOptions.getDeclaredFields()).anyMatch{ f -> f.getName().equals('TCP_KEEPCOUNT') }) {
-            Method getOptionMethod = Socket.getMethod('getOption', SocketOption)
-            getOptionMethod.invoke(socket, ExtendedSocketOptions.getDeclaredField('TCP_KEEPCOUNT').get(null)) == 9
-            getOptionMethod.invoke(socket, ExtendedSocketOptions.getDeclaredField('TCP_KEEPIDLE').get(null)) == 120
-            getOptionMethod.invoke(socket, ExtendedSocketOptions.getDeclaredField('TCP_KEEPINTERVAL').get(null)) == 10
+            Method getOptionMethod
+            try {
+                getOptionMethod = Socket.getMethod('getOption', SocketOption)
+            } catch (NoSuchMethodException e) {
+                // ignore, the `Socket.getOption` method was added in Java SE 9 and does not exist in Java SE 8
+                getOptionMethod = null
+            }
```

Comment:
This code assumed that if `ExtendedSocketOptions.TCP_KEEPCOUNT` is available, then so is `Socket.getOption`. That's not actually the case:

- `ExtendedSocketOptions.TCP_KEEPCOUNT` (not part of the Java SE) was added in Oracle JDK 8: https://docs.oracle.com/javase/8/docs/jre/api/net/socketoptions/spec/jdk/net/ExtendedSocketOptions.html#TCP_KEEPCOUNT
- `Socket.getOption` was added in the Java SE 9.
- So when I run `sdk use java 17.0.10-librca && ./gradlew -PjavaVersion=8 clean :driver-core:test` locally, `ExtendedSocketOptions.TCP_KEEPCOUNT` is present, while `Socket.getOption` is not, which leads to a failure.
- Why is this not a problem in Evergreen? My guess is that the JDK 8 it uses does not have `ExtendedSocketOptions.TCP_KEEPCOUNT`.